rosetta-source/src/main/rosetta/base-staticdata-asset-common-type.rosetta

```diff
@@ -134,29 +161,19 @@ type CommodityReferenceFramework: <"Specifies the type of commodity.">
 
 type DigitalAsset extends AssetBase:  <"An Asset that exists only in digital form, eg Bitcoin or Ethereum, that is not backed by other Assets; excludes the digital representation of other Assets, eg coins or Tokenised assets.">
 
-type ListedDerivative extends AssetBase: <"A securitized derivative on another asset.">
-    isExchangeTraded boolean (0..1) <"Identifies whether the derivative is traded on an exchange.">
-    exchange string (0..1) <" Identifies the exchange where the asset is listed, using the scheme definition http://www.fpml.org/coding-scheme/external/exchange-id-MIC-1-0">
-        [metadata scheme]
+type ListedDerivative extends InstrumentBase: <"A securitized derivative on another asset.">
     exerciseTerms ExerciseTerms (1..1) <"Specifically the ExerciseDate.">
     optionType OptionTypeEnum (0..1) <"The type of option, ie Put or Call. Left empty if it is a Future.">
     strike OptionStrike (0..1) <"Specifies the strike of the option.">
 
```

Comment:
Quite uncomfortable with the inclusion of these "product" attributes in the `ListedDerivative` type. These should be the preserve of `Economicterms`. This also breaks the namespace hierarchy.

If some "simple" attributes are really necessary, consider using basic types such as `date` or `number`.

Reply:
The rationale is that an identifier alone is not sufficient to identify a ListedDerivative contract.  Usually the identifier identifies the root contract, eg the E-mini S&P 500 future is "ES".  The expiration date needs to be added as an additional attribute.  For Options, one also needs a put/call and strike price to uniquely identify the contract.

However, I understand about the hierarchy.  So will look at making them base types.

Reply:
Ok I understand. Yes in this case make them as basic types. And `OptionTypeEnum` should be moved to the "base" namespace.

For the strike though, I thought that you could only choose from a given set of values, otherwise they wouldn't be fungible. Isn't that meant to be captured by another identifier that specifies that contract beyond the root?

Reply:
Latest PR has as basic types, but I will update to use `OptionTypeEnum` in updated namespace.

Strike Price is usually just a string or number as the contracts change a lot; eg in FIXML:

`<Instrmt SecType="OPT" CFICode="OPXXXX" Symbol="ES" MMY="202406" Exch="CME" PutCall="0" StrikePx="4400" />`